plugins/code-coverage/src/components/FileExplorer/FileContent.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { FC, useEffect } from 'react';
+import { useApi } from '@backstage/core-api';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { useAsync } from 'react-use';
+import { codeCoverageApiRef } from '../../api';
+import { Progress } from '@backstage/core';
+import { Alert } from '@material-ui/lab';
+import { makeStyles, Paper } from '@material-ui/core';
+import { highlightLines } from './Highlighter';
+import { FileCoverage } from './FileExplorer';
+import CoverageRow from './CoverageRow';
+
+type Props = {
+  filename: string;
+  coverage: FileCoverage;
+};
+
+const getContent = async (value: any) => {
+  let blob = value;
+  if (value instanceof ArrayBuffer || ArrayBuffer.isView(value)) {
+    blob = new Blob([value], { type: 'application/octet-stream' });
+  }
+  return value;
+};
+
+const useStyles = makeStyles(theme => ({
+  paper: {
+    margin: 'auto',
+    top: '2em',
+    width: '80%',
+    backgroundColor: theme.palette.background.paper,
+    border: '2px solid #000',
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 4, 3),
+    overflow: 'scroll',
+  },
+  coverageFileViewTable: {
+    borderSpacing: '0px',
+    width: '80%',
+    marginTop: theme.spacing(2),
+  },
+}));
+
+type FormattedLinesProps = {
+  highlightedLines: string[];
+  lineHits: Record<number, number>;
+};
+const FormattedLines: FC<FormattedLinesProps> = ({
+  highlightedLines,
+  lineHits,
+}: FormattedLinesProps) => {
+  return (
+    <>
+      {highlightedLines.map((lineContent, idx) => {
+        const line = idx + 1;
+        return (
+          <CoverageRow
+            key={line}
+            lineNumber={line}
+            lineContent={lineContent}
+            lineHits={lineHits[line]}
+          />
+        );
+      })}
+    </>
+  );
+};
+
+export const FileContent = ({ filename, coverage }: Props) => {
+  const { entity } = useEntity();
+  const codeCoverageApi = useApi(codeCoverageApiRef);
+  const { loading, error, value } = useAsync(
+    async () =>
+      await codeCoverageApi.getFileContentFromEntity(
+        {
+          kind: entity.kind,
+          namespace: entity.metadata.namespace || 'default',
+          name: entity.metadata.name,
+        },
+        filename,
+      ),
```

Comment:
```suggestion
      ), [entity]
```

so it reloads properly when the route changes